Xamarin.Forms.Platform.iOS/Extensions/BrushExtensions.shared.cs

```diff
@@ -63,5 +63,35 @@ static NSNumber[] GetCAGradientLayerLocations(List<GradientStop> gradientStops)
 				return locations;
 			}
 		}
+
+		static CGColor[] GetCAGradientLayerColors(List<GradientStop> gradientStops)
+		{
+			if (gradientStops == null || gradientStops.Count == 0)
+				return new CGColor[0];
+
+			CGColor[] colors = new CGColor[gradientStops.Count];
+
+			int index = 0;
+			foreach (var gradientStop in gradientStops)
+			{
+				if (gradientStop.Color == Color.Transparent)
+				{
+					var color = gradientStops[index == 0 ? index + 1 : index - 1].Color;
```

Comment:
Can you help me understand the logic here?

Reply:
If the GradientStop color is transparent, use the next one  Color with Alpha 0. In other case, the effect is not correct because use a Clear default color and can see a black color in the gradient from one to other color. 
This article explain it well: https://betterprogramming.pub/the-proper-way-of-creating-a-transparent-gradient-layer-in-ios-b082daa866b1